### PR TITLE
Reformat code with Black

### PR DESCRIPTION
**Pull request by mryab**
Adjusted the default configuration with 119 characters per line (this is the highest number allowed by GitHub PR view window). 

Also, added pyproject.toml for configuring tools such as black/isort. However, compiling libp2p from scratch conflicts with the PEP517 build system backend, which is enabled by pyproject.toml: specifically, supplying `build_py` makes pip compile all dependencies from scratch (see https://github.com/pypa/pip/issues/4118), which significantly increases the build time. To mitigate this issue, for now I supplied the `--no-use-pep517` option to pip (see https://github.com/learning-at-home/hivemind/pull/274/commits/5ef79dace8148990c25d370c6c53c7308a907929 — without this change, the build works correctly, but we need to remove the timeout). This will likely be documented in https://github.com/learning-at-home/hivemind/pull/307, along with other considerations related to libp2p installation

### CODE REVIEW

File: hivemind/dht/routing.py [hunk 1]
```diff
@@ -11,7 +11,13 @@
...
 from hivemind.utils import MSGPackSerializer, get_dht_time
 
-DHTKey, Subkey, DHTValue, BinaryDHTID, BinaryDHTValue, = Any, Any, Any, bytes, bytes
+DHTKey, Subkey, DHTValue, BinaryDHTID, BinaryDHTValue, = (
+    Any,
+    Any,
+    Any,
+    bytes,
+    bytes,
+)
```
Comment (borzunov) [inline hivemind/dht/routing.py:20]:
```suggestion
DHTKey = Subkey = DHTValue = Any
BinaryDHTID = BinaryDHTValue = bytes
```

File: hivemind/dht/routing.py [hunk 2]
```diff
@@ -134,7 +143,9 @@ def get_nearest_neighbors(
                 while right_index < len(self.buckets) and self.buckets[right_index].upper <= current_upper:
                     for node_id, endpoint in self.buckets[right_index].nodes_to_endpoint.items():
                         heapq.heappush(candidates, (query_id.xor_distance(node_id), node_id, endpoint))
-                    right_index += 1  # note: we may need to add more than one bucket if they are on a lower depth level
+                    right_index += (
+                        1  # note: we may need to add more than one bucket if they are on a lower depth level
+                    )
```
Comment (borzunov) [inline hivemind/dht/routing.py:148]:
```suggestion
                    right_index += 1
                    # note: we may need to add more than one bucket if they are on a lower depth level
```

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -56,7 +56,7 @@ jobs:
           pip install -r requirements-dev.txt
       - name: Build hivemind
         run: |
-          pip install . --global-option=build_py --global-option="--buildgo"
+          pip install . --global-option=build_py --global-option="--buildgo" --no-use-pep517
```
Comment (borzunov) [inline .github/workflows/run-tests.yml:59]: What does it do?

**Review by borzunov — APPROVED** [review]

**Review by justheuristic — APPROVED** [review]

File: hivemind/averaging/training.py [hunk 4]
```diff
@@ -86,15 +95,17 @@ def step(self, data_lock: Optional[Lock] = None, wait: bool = True, **kwargs):
...
-                            local_tensor[...] += averaged_tensor.to(dtype=local_tensor.dtype,
-                                                                    device=local_tensor.device) - \
-                                                 old_local_tensor.to(dtype=local_tensor.dtype,
-                                                                     device=local_tensor.device)
+                        for averaged_tensor, local_tensor, old_local_tensor in zip(
+                            averaged_tensors, local_tensors, old_local_tensors
+                        ):
+                            local_tensor[...] += averaged_tensor.to(
+                                dtype=local_tensor.dtype, device=local_tensor.device
+                            ) - old_local_tensor.to(dtype=local_tensor.dtype, device=local_tensor.device)
```
Comment (yhn112) [inline hivemind/averaging/training.py:103]:
```suggestion
                            local_tensor[...] += averaged_tensor.to(local_tensor.dtype).to(local_tensor.device)
                            local_tensor[...] -= old_local_tensor.to(local_tensor.dtype).to(local_tensor.device)
```

File: hivemind/averaging/training.py [hunk 5]
```diff
@@ -86,15 +95,17 @@ def step(self, data_lock: Optional[Lock] = None, wait: bool = True, **kwargs):
...
                     else:
                         for averaged_tensor, local_tensor in zip(averaged_tensors, local_tensors):
-                            local_tensor[...] = averaged_tensor.to(dtype=local_tensor.dtype, device=local_tensor.device)
+                            local_tensor[...] = averaged_tensor.to(
+                                dtype=local_tensor.dtype, device=local_tensor.device
+                            )
```
Comment (yhn112) [inline hivemind/averaging/training.py:108]:
```suggestion
                            local_tensor[...] = averaged_tensor.to(local_tensor.dtype).to(local_tensor.device)
```
Maybe now you think that I'm a moron. Sorry.

**Review by yhn112 — APPROVED** [review]

**Pull request merged**